cli/src/commands/git/push.rs

```diff
@@ -369,6 +407,13 @@ fn validate_commits_ready_to_push(
     } else {
         Box::new(|_: &CommitId| Ok(false))
     };
+    let mut sign_settings = settings.sign_settings();
+    sign_settings.behavior = if settings.get_bool("git.sign-on-push").unwrap_or(false) {
```

Comment:
nit: we usually put the default value in `misc.toml` (or other config file) so we can expect a valid value to be available and do `get_bool(...)?` when we read it